app/scripts/HiGlassTrackComponent.js

```diff
@@ -1,69 +1,54 @@
 
 /**
- * A barebones implementation of a HiGlass view used to display a 2d track.
- * Multiple layers can be used by specifying a `combined` track in `trackConfig`.
- *
- * Defaults to having no label and no colorbar.
- *
- * If the aspect ratio defined by the parameter bounds doesn't match the aspect
- * ration of the enclosing element, the x bounds will be respected and the y
- * y coordinates will be truncated.
- *
- * @param {element|HTMLElement} The element to attach the viewer to
- * @param {[xMin, xMax, yMin, yMax]|Array} The bounds of the track
- * @param {trackConfig: Object} The standard HiGlass track definition type
- * @returns {Object} A {id, hgApi} object
- */
-export const trackViewer = (element, [xMin, xMax, yMin, yMax], trackConfig) => {
-  if (!trackConfig.options.colorbarPosition) {
-    trackConfig.options.colorbarPosition = 'hidden';
-  }
-  if (!trackConfig.options.labelPosition) {
-    trackConfig.options.labelPosition = 'hidden';
-  }
-  const id = 'arbitary-id';
-  const viewConfig = {
-    editable: false,
-    zoomFixed: false,
-    views: [
-      {
-        uid: id,
-        initialXDomain: [xMin, xMax],
-        initialYDomain: [yMin, yMax],
-        tracks: {
-          center: [trackConfig],
-        },
-        layout: {
-          w: 12,
-          h: 12,
-          x: 0,
-          y: 0,
-          moved: false,
-          static: false,
-        },
-      },
-    ],
-  };
-  const hgApi = viewer(element, viewConfig, { bounded: true });
-  return { id, hgApi };
-};
-
-/**
- * React component wrapper around trackViewer.
+ * React component for single 2D HiGlass tracks.
  * Accepts x, y, width, and height props, in addition to trackConfig,
  * so HiGlass can be used to provide background imagery for Deck.gl.
  */
 export default class HiGlassTrackComponent extends React.Component {
   constructor(props) {
     super(props);
-    this.id = `id-${Math.random()}`;
-  }
 
-  componentDidMount() {
-    this.initTrackViewer();
+    const {
+      trackConfig, x, y, width, height
+    } = props;
+    const xMin = x;
+    const xMax = x + width;
+    const yMin = y;
+    const yMax = y + height;
+    if (!trackConfig.options.colorbarPosition) {
+      trackConfig.options.colorbarPosition = 'hidden';
+    }
+    if (!trackConfig.options.labelPosition) {
+      trackConfig.options.labelPosition = 'hidden';
+    }
+    this.viewUid = 'UID-placeholder';
+    this.viewConfig = {
+      editable: false,
+      zoomFixed: false,
+      views: [
+        {
+          uid: this.viewUid,
+          initialXDomain: [xMin, xMax],
+          initialYDomain: [yMin, yMax],
+          tracks: {
+            center: [trackConfig],
+          },
+          layout: {
+            w: 12,
+            h: 12,
+            x: 0,
+            y: 0,
+            moved: false,
+            static: false,
+          },
+        },
+      ],
+    };
+    this.options = { bounded: true };
+    this.HgcRef = React.createRef();
```

Comment:
```suggestion
    this.hgcRef = React.createRef();
```
camelCase for something that isn't a class, no?